### PR TITLE
misc: `go vet` fixes from go tip

### PR DESCRIPTION
**Pull request by tamird**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/5638)

### CODE REVIEW

File: rpc/context.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func NewServer(ctx *Context) *grpc.Server {
 // Context contains the fields required by the rpc framework.
 type Context struct {
 	// Embed the base context.
-	base.Context
+	*base.Context
```
Comment (petermattis) [inline rpc/context.go]: I'm curious what vet was complaining about here.

Comment (tamird) [inline rpc/context.go]: it wasn't complaining about anything here, but it complained about the dereference in the constructor copying a mutex.

Comment (petermattis) [inline rpc/context.go]: Ah, got it.

**Comment by petermattis** [comment]
LGTM

**Pull request merged**